backend/src/middleware/auth.middleware.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable, NestMiddleware, UnauthorizedException } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import axios from 'axios'
+import * as jwt from 'jsonwebtoken'
+
+
+@Injectable()
+export class AuthMiddleware implements NestMiddleware {
+
+    use(req: Request, res: Response, next: NextFunction) {
+    
+        var access_token = req.cookies['accessToken']
+        var refresh_token = req.cookies['refreshToken']
+        
+        if (!access_token || !refresh_token) {
+            throw new UnauthorizedException("Missing Token");
+        }
+
+        var result = this.decode_token(access_token);
+
+        if (result.payload && !result.expired) {
+            
+        } else if (result.expired) {
+            access_token = this.refresh_token(refresh_token);
```

Comment:
Du rufst hier eine asynchrone Funktion auf, daher musst du sie awaiten oder mit `.then()` arbeiten.
Wie du es hier hast, hast du in der variable `access_token` ein unaufgelöstes Promise.
Bitte noch einmal prüfen.